src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectPostConfigureOptions.cs

```diff
@@ -30,6 +30,11 @@ public OpenIdConnectPostConfigureOptions(IDataProtectionProvider dataProtection)
         /// <param name="options">The options instance to configure.</param>
         public void PostConfigure(string name, OpenIdConnectOptions options)
         {
+            if (options.MaxAge != null && options.MaxAge.Value < TimeSpan.Zero) 
+            {
+                throw new InvalidOperationException("MaxAge must not be a negative TimeSpan.");                
```

Comment:
This actually belongs over here:
https://github.com/aspnet/Security/blob/442df16e265ab1955d7b8468cf1d8b38b8d915c7/src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectOptions.cs#L87

Reply:
Well, both places are fine 'cause we do validation in the initializer too. I suggested removing the `Validate` method from the options, but @HaoK wasn't okay with that... :trollface:

Reply:
OpenIdConnectPostConfigureOptions is for delayed initialization, not basic args validation.

Reply:
Yep, Validate on the Options class itself should be the source of truth for validation

Reply:
> OpenIdConnectPostConfigureOptions is for delayed initialization, not basic args validation.
 
I still think the two places should be consolidated. There's no reason to repeat the same validation checks for each request again and again.